### ISSUE TITLE
Sources generated before #1208 might not compile with newer runtime

### ISSUE THREAD

**Issue by thomasvl** [apple/swift-protobuf#1212]
Stumbled into a case where some sources generated before #1208 landed fail to compile when compiled with the updated library. Seems to come down to the Data/Oneof support since we are adding sendable annotations in the new generated sources, the old generated sources don't have them, but the new runtime tries to claim all Message will be sendable, but without those annotations the compiler can't confirm it.

**Comment by thomasvl** [comment]
@zwaldowski fyi

**Comment by thomasvl** [comment]
Repo is easy enough:

1. Checkout current _main_
2. `git revert aaead67360cd4d046aabae0f57bac4b5e15d7628`
3. `swift build`

Get a bunch of issues like:

```
/Users/thomasvl/src/github/swift-protobuf/Sources/SwiftProtobuf/struct.pb.swift:127:14: error: stored property 'kind' of 'Sendable'-conforming struct 'Google_Protobuf_Value' has non-sendable type 'Google_Protobuf_Value.OneOf_Kind?'
  public var kind: Google_Protobuf_Value.OneOf_Kind? = nil
             ^
```

**Comment by tbkka** [comment]
We've bumped the compatibility version before and the world did not actually fall apart.  ;-)

It's nice to avoid bumping the compatibility version when possible, but it doesn't seem possible.

**Comment by thomasvl** [comment]
We might not have bumped it since hitting 1.0.  What are the semver rules for something like this?  No, we aren't changing any apis, but all the generated code won't compile.

**Comment by Lukasa** [comment]
The best fix is probably to add a new protocol (`_ConcurrentMessage`?) that refines `Message` and `Sendable`, and conform message types in the codegen to that as well. That means that if you regen the code you get Sendable everywhere, if you don’t you don’t. You _dont_ automatically get `Message: Sendable` for other generics, but we really can’t avoid the code break there unless we can prevent people updating the runtime.

**Comment by zwaldowski** [comment]
> The best fix is probably to add a new protocol (`_ConcurrentMessage`?) that refines `Message` and `Sendable`, and conform message types in the codegen to that as well. That means that if you regen the code you get Sendable everywhere, if you don’t you don’t. You _dont_ automatically get `Message: Sendable` for other generics, but we really can’t avoid the code break there unless we can prevent people updating the runtime.

I'm like this as a solution, but that would break in the other direction, right? Newer code-gen, older support library. I don't know that y'all care about that as a use case, but it's still a compatibility break. My team sometimes runs into problems with that because someone can acquire a new version of `protoc-gen-swift` without meaning to, f.ex., they get a replacement machine.

---

If we're not fully aligned on next steps, I'm leaning towards the "stop the bleeding" approach, because working in a generic context isn't mission-critical. I just want to do what's best for the ecosystem. :)

**Comment by tbkka** [comment]
Based on the discussion so far, I propose:
* _Not_ bumping the compatibility version
* Generating the conformances on specific messages and enums
* Removing the `Message: Sendable` and `Enum: Sendable` conformances
* Recording a new Issue to add those conformances at some time in the future. (we have a few other "2.0" changes floating about)

**Comment by zwaldowski** [comment]
Still happy to go another way, but opened #1213 with that low-effort mitigation.

**Comment by thomasvl** [comment]
Should be fixed via #1213
